samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -276,9 +278,66 @@ def _generate_custom_makefile(
     cfn_resources: dict
         CloudFormation resources
     output_directory_path: str
-        the output directory path to write the generated metadata and makefile
+        the output directory path to write the generated makefile
+    terraform_application_dir: str
+        the terraform project root directory
+    """
+
+    # create output directory if it doesn't exist
+    if not os.path.exists(output_directory_path):
+        os.makedirs(output_directory_path, exist_ok=True)
+
+    # copy copy_terraform_built_artifacts.py script into output directory
+    shutil.copy("../copy_terraform_built_artifacts", output_directory_path)
```

Comment:
not sure on how this would be handled in windows, always have to be careful with paths to ensure they are cross platfrom.

Reply:
good call out to be careful just in case, I think my latest commit should address it